src/components/AdminSettings/TurnServer.vue

```diff
@@ -234,11 +235,19 @@ export default {
 			}
 
 			// Create a PeerConnection with no streams, but force a m=audio line.
-			const config = {
-				iceServers: [
-					iceServer,
-				],
-				iceTransportPolicy: 'relay',
+			// Only items with non-empty urls may occur in iceServers according to WebRTC spec.
+			if (urls.length) {
+				const config = {
+					iceServers: [
+						iceServer,
+					],
+					iceTransportPolicy: 'relay',
+				}
+			} else {
+				const config = {
+					iceServers: [],
+					iceTransportPolicy: 'relay',
+				}
```

Comment:
If I am not mistaken this should not be needed, because if any of the elements that compose the urls is empty the method will [return before even trying to establish a connection](https://github.com/nextcloud/spreed/blob/d368b50176afc233a9448a4f8478a10eaff19af2/src/components/AdminSettings/TurnServer.vue#L215). However, it seems to be stuck because [the loading spinner is not reset before returning](https://github.com/nextcloud/spreed/blob/d368b50176afc233a9448a4f8478a10eaff19af2/src/components/AdminSettings/TurnServer.vue#L209).

I think that a better approach would be not to change the code related to the ice servers and reset the loading spinner instead before returning if any component is empty.

Reply:
Fixed in 3eaf05e87813f44a1b21d12fccf316c309ef40c0.